### PR TITLE
fix: show the actual deno cli version in `Deno.version`

### PR DESCRIPTION
**Pull request by nyannyacha**
## What kind of change does this PR introduce?

Bug fix

## Description

```typescript
console.log(Deno.version);
// Output (Before / local build):
// {
//     "deno": "supabase-edge-runtime-0.1.0",
//     "v8": "11.6.189.12",
//     "typescript": "5.1.6"
// }
// Output (Before / prod):
// {
//     "deno": "supabase-edge-runtime-",
//     "v8": "11.6.189.12",
//     "typescript": "5.1.6"
// }

// Output (After / local build):
// {
//     "deno": "supabase-edge-runtime-0.1.0 (compatible with Deno v1.39.2)",
//     "v8": "11.6.189.12",
//     "typescript": "5.1.6"
// }

// Output (After / prod):
// {
//     "deno": "supabase-edge-runtime-[release version here] (compatible with Deno v1.39.2)",
//     "v8": "11.6.189.12",
//     "typescript": "5.1.6"
// }

console.log(DENO_VERSION);
// Output:
// 1.39.2
```

This PR indirectly relevant with the [question](https://discord.com/channels/839993398554656828/1202913385067577364) that discussed in Discord.

Reported-by: @paul-brenner

### CODE REVIEW

**Review by andreespirela — APPROVED** [review]
LGTM!

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -61,7 +61,7 @@ jobs:
           cache-from: type=gha
           cache-to: type=gha,mode=max
           build-args: |
-            GIT_V_VERSION=${{ steps.semantic.outputs.new_release_version }}
+            GIT_V_VERSION=${{ needs.release.outputs.version }}
```
Comment (nyannyacha) [inline .github/workflows/release.yml:64]: cc @laktek @andreespirela

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -103,6 +103,8 @@ jobs:
           platforms: linux/${{ env.arch }}
           tags: ${{ steps.meta.outputs.tags }}
           no-cache: true
+          build-args: |
+            GIT_V_VERSION=${{ needs.release.outputs.version }}
```
Comment (nyannyacha) [inline .github/workflows/release.yml:107]: cc @laktek @andreespirela

File: .github/workflows/release.yml (hunk 1 above)
Comment (nyannyacha) [reply .github/workflows/release.yml:64]: https://github.com/orgs/supabase/discussions/21000

**Review by laktek — APPROVED** [review]

**Pull request merged**